contracts/contracts/MACI.sol

```diff
@@ -101,6 +101,12 @@ contract MACI is IMACI, DomainObjs, Params, Utilities, Ownable {
     stateAq = new AccQueueQuinaryBlankSl(STATE_TREE_SUBDEPTH);
     stateAq.enqueue(BLANK_STATE_LEAF_HASH);
 
+    // because we add a blank leaf we need to count one signup
+    // so we don't allow max + 1
+    unchecked {
+      numSignUps++;
+    }
```

Comment:
Shouldn't users expect numSignUps to reflect the actual number of signups without having to adjust for the BLANK_STATE_LEAF by subtracting one manually? This could lead to misleading data representation

Reply:
Fair point, although I think it depends on what you mean by "users" & options of how to represent that data to users. For instance, a frontend integration could easily account for this by adjusting what the data they display in the application

An alternative might be to not count the blank leaf as a signup, but to restrict the total number of signups by 1